controllers/etcdpeer_controller.go

```diff
@@ -162,6 +163,53 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer) appsv1.ReplicaSet {
 	}
 }
 
+func pvcForPeer(peer *etcdv1alpha1.EtcdPeer) *corev1.PersistentVolumeClaim {
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      peer.Name,
+			Namespace: peer.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(peer, etcdv1alpha1.GroupVersion.WithKind("EtcdPeer")),
```

Comment:
If you set an owner reference, won't the PVC get deleted if you delete the `EtcdPeer`?

Reply:
Yes, and I think this is the same behaviour as the StatefulSet controller, but the dynamically provisioned PV will not be deleted, so long as you configure the StorageClass reclaimPolicy. See: 
 * https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy
 * https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#stable-storage

Reply:
Additionally, if I don't set the OwnerReference here, we won't be be able to  watch PVCs using the `Owns()` method below.

Reply:
Ah yes, my mistake.

Reply:
If we don't delete the PVC at all, then when the EtcdPeer is re-created it will 'adopt' the old PVC.